werror.go

```diff
@@ -232,90 +237,113 @@ func (e *werror) Cause() error {
 	return e.cause
 }
 
+// stack returns the Stacktracer for this error or nil if there is none.
+func (e *werror) StackTrace() StackTrace {
+	return e.stack
+}
+
+// Message returns the message string for this error or nil if there is none.
+func (e *werror) Message() string {
+	return e.message
+}
+
 func (e *werror) SafeParams() map[string]interface{} {
-	safe, _ := ParamsFromError(e)
+	safe := make(map[string]interface{})
+	for k, v := range e.params {
+		if v.safe {
+			safe[k] = v.value
+		}
+	}
 	return safe
 }
 
 func (e *werror) UnsafeParams() map[string]interface{} {
-	_, unsafe := ParamsFromError(e)
+	unsafe := make(map[string]interface{})
+	for k, v := range e.params {
+		if !v.safe {
+			unsafe[k] = v.value
+		}
+	}
 	return unsafe
 }
 
 // Format formats the error using the provided format state. Delegates to stored error.
 func (e *werror) Format(state fmt.State, verb rune) {
+	FormatWerror(e, state, verb)
+}
+
+// FormatWerror formats a Werror using the provided format state.
+func FormatWerror(err Werror, state fmt.State, verb rune) {
```

Comment:
This allows future `Werror` implementations to share the formatting logic with the existing `werror` type, which should help ensure consistency.

Reply:
nit: How about making this just `Format` given it's in the `werror` pkg and it takes in a `Werror` as the first param.
```go
werror.Format(...)
// vs
werror.FormatWerror(...)
```

Reply:
👍 Done.